test/extended/storage/mutable_csinode_allocatable.go

```diff
@@ -0,0 +1,678 @@
+	defer g.GinkgoRecover()
+	var (
+		ctx                                 = context.Background()
+		oc                                  = exutil.NewCLI("mutable-csinode-allocatable-test")
+		originalCSIDriverUpdatePeriod       *int64
+		targetWorkerNode                    string
+		ec2Client                           *ec2.EC2
+		originalAllocatableCount            int32
+		originalCVOReplicas                 *int32
+		targetWorkerNodeAttachedVolumeCount int32
+	)
+
+	g.BeforeEach(func() {
+		// TODO: remove the check when MutableCSINodeAllocatableCount is supported for GA
+		if !exutil.IsTechPreviewNoUpgrade(ctx, oc.AdminConfigClient()) {
+			g.Skip("skipping, this feature is only supported on TechPreviewNoUpgrade clusters")
+		}
+
+		// Skip if not AWS platform
+		if !e2e.ProviderIs("aws") {
+			g.Skip("skipping, this test is only expected to work with AWS clusters")
+		}
+
+		// TODO: After https://issues.redhat.com/browse/OCPBUGS-65972 is resolved, remove the check,
+		// we can use the change the clustercsidriver Unmanaged way which will be also suitable for hosted clusters.
+		// Skip if on hosted clusters
+		controlPlaneTopology, err := exutil.GetControlPlaneTopology(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if *controlPlaneTopology == configv1.ExternalTopologyMode {
+			g.Skip("skipping, this test is only expected to work with standalone clusters.")
+		}
+
+		// Check to see if we have Storage enabled
+		isStorageEnabled, err := exutil.IsCapabilityEnabled(oc, configv1.ClusterVersionCapabilityStorage)
+		if err != nil || !isStorageEnabled {
+			g.Skip("skipping, this test is only expected to work with storage enabled clusters")
+		}
+
+		isMicroShift, err := exutil.IsMicroShiftCluster(oc.AdminKubeClient())
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isMicroShift {
+			g.Skip("MutableCSINodeAllocatableCount tests are not supported on MicroShift")
+		}
+		// TODO: After https://issues.redhat.com/browse/OCPBUGS-65972 is resolved,
+		// we can use the change the clustercsidriver Unmanaged way which will be even better.
+		g.By("Scaling down cluster-version-operator, cluster-storage-operator, and aws-ebs-csi-driver-operator")
+		// Scale down cluster-version-operator
+		cvoDeployment, err := oc.AdminKubeClient().AppsV1().Deployments("openshift-cluster-version").Get(ctx, "cluster-version-operator", metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred(), "failed to get cluster-version-operator deployment")
+		originalCVOReplicas = cvoDeployment.Spec.Replicas
+
+		zero := int32(0)
+		_, err = e2edeployment.UpdateDeploymentWithRetries(oc.AdminKubeClient(), "openshift-cluster-version", "cluster-version-operator", func(d *appsv1.Deployment) {
+			d.Spec.Replicas = &zero
+		})
+		o.Expect(err).NotTo(o.HaveOccurred(), "failed to scale down cluster-version-operator")
+		e2e.Logf("Successfully scaled down cluster-version-operator")
```

Comment:
Extract it into a function and call it to scale CVO + CSO + AWS operator down and up.